path.go

```diff
@@ -278,10 +279,12 @@ func (c *compiler) parseFilter(path string) filter {
 		}
 	}
 
-	// Filter contains [@attr], [N] or [tag]
+	// Filter contains [@attr], [N], [tag] or [text()]
 	switch {
 	case path[0] == '@':
 		return newFilterAttr(path[1:])
+	case strings.HasPrefix(path, "text()"):
```

Comment:
I don't think you want HasPrefix here.  I think you want to fully compare path to "text()".